### PR TITLE
Update AUTHORS and .mailmap

### PR DESCRIPTION
**Pull request by robn**
### Motivation and Context

I wanted my name in the AUTHORS file, and I could not just be normal and type my damn name.

### Description

This adds a program to be run manually. It sifts through the git history, trying to discover the "best" name for a committer by collecting up every email address and name they've ever used and using a bunch of heuristics to pick the nicest one.

I'm adding a `.mailmap` to connect names that can't be correlated by name or address matching. in particular this includes `@users.noreply.github.com`, except in a couple of cases where there's no other way.

The `AUTHORS` file is treated as the final source of truth, regardless of what's in the commit history.

The "removed" lines in AUTHORS are just reordering; I've checked them all and they're all just moved, not changed or removed.

The program is written in simple ancient Perl, so it should run anywhere. I have no particular guidance on when to run it; probably just ad-hoc is fine unless we start getting a regular stream of new contributors.

### How Has This Been Tested?

No testing as such. I just kept adjusting things until I got the answers I wanted.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by behlendorf — APPROVED** [review]
This all works as advertised.   Thanks for going above and beyond so we can update this file and have some reasonable means up keeping it current!

File: scripts/update_authors.pl [hunk 1]
```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env perl
```
Comment (behlendorf) [inline scripts/update_authors.pl:1]: Let's add this script to `dist_noinst_SCRIPTS` section of `scripts/Makefile.am`.

**Comment by robn** [comment]
Thanks for humouring me :D

**Review by mcmilk — APPROVED** [review]
I like the idea and also the implementation is well done.

Just a small note: should we implement some simple email hiding to stop dumb spambots from getting all the email adresses?
Maybe sth. like this: `Real Name <user |a| example.org>`

**Comment by robn** [comment]
They're all already visible in the commit history, which is easily readable via any git web interface. Also, (with my old email sysadmin hat on) deobfuscating email addresses isn't hard to do, and isn't a huge source of addresses for spammers anyway.

For myself, I actually want my real address in the list, since it helps search engines connect the bits of my online presence, which helps people find me and send work my way.

All that said, many people are quite reasonably wary when it comes to how their names and other identifiers are presented, and I'm definitely in a privileged position, so I'm not going to argue if anyone feels especially strongly about it.

**Comment by mcmilk** [comment]
This was _Just a small note_ - and for myself, I have also no problem with it ;-)
@robn - thanks for your thoughts on this.

**Pull request closed without merging**